### PR TITLE
Include oe_verify_evidence() in host_verify.h and used to verify evidence 

### PR DESCRIPTION
**Pull request by shants**
Signed-off-by: Shantanu Sharma <shanshar@microsoft.com>

### CODE REVIEW

**Review by yentsanglee — COMMENTED** [review]
include/openenclave/attestation/sgx/evidence.h has to be included in host_verify package so caller can specify SGX specific UUIDs in oe_verify_evidnece(). Check out include/CMakeLists.txt file.

File: samples/host_verify/host.c [hunk 1]
```diff
@@ -110,12 +115,23 @@ oe_result_t verify_report(
...
-        result = oe_verify_remote_report(
+        oe_verifier_initialize();
+        result = oe_verify_evidence(
+            &_uuid_legacy_report_remote,
             report_data,
             report_file_size,
-            endorsement_data,
-            endorsement_file_size,
-            NULL);
+            NULL,
```
[the remark this answers is not in the document]
Comment (shants) [reply samples/host_verify/host.c]: Added endorsement_data and endorsement_file_size.

**Review by ryanhsu19 — COMMENTED** [review]
IMO since line 41 covers `openenclave/attestation/sgx/evidence.h` (which will be packed into both OE package and `oehostverify` package), line 32 can be removed.

**Review by ryanhsu19 — APPROVED** [review]

File: samples/host_verify/host.c [hunk 2]
```diff
@@ -110,12 +115,23 @@ oe_result_t verify_report(
...
+            0,
+            &claims,
+            &claims_length);
+    }
+
+    printf("Printing Claim Names\n ");
+    for (size_t j = 0; j < claims_length; j++)
+    {
+        printf("%s \n", claims[j].name);
```
Comment (qiucwang) [inline samples/host_verify/host.c:134]: Maybe we can show more info in a sample here? Like `printf("%s(%lu)\n", claims[j].name, claims[j].value_size)`.
Just a personal thought.

Comment (shants) [reply samples/host_verify/host.c]: Sure, that should be fine. Added it now

File: include/openenclave/host_verify.h [hunk 3]
```diff
@@ -122,6 +122,68 @@ oe_result_t oe_verify_attestation_certificate_with_evidence(
     oe_verify_claims_callback_t claim_verify_callback,
     void* arg);
 
+/**
```
Comment (dthaler) [inline include/openenclave/host_verify.h:125]: I don't like the idea of having duplicate definitions in different files.
Can we have a header file organization that reuses a common header that is included by both host_verify.h and attestation/verifier.h that includes everything that's identical in both?

**Review by yentsanglee — APPROVED** [review]
Looks good. With new API, host_verify.h is not needed.

File: include/openenclave/host_verify.h (hunk 3 above)
Comment (shants) [reply include/openenclave/host_verify.h]: Thanks for highlighting the duplication. openenclave/attestation/verifier.h is included in oehostverify. Removed the functions  oe_verify_evidenece() and oe_verify_attestation_certificate_with_evidence() from host_verify.h

**Comment by shants** [comment]
bors retry

**Pull request merged**